tests/setup-softhsm.sh

```diff
@@ -137,10 +137,14 @@ ca_sign() {
     CN=$3
     KEYID=$4
     let "SERIAL+=1"
+    # BSD sed expects a space between -i and the backup extension, GNU sed
+    # expects the extension right after the -i argument. Portable scripts thus
+    # can't use sed -i.
     sed -e "s|cn = .*|cn = $CN|g" \
         -e "s|serial = .*|serial = $SERIAL|g" \
         -e "/^ca$/d" \
-        -i ${TMPPDIR}/cert.cfg
```

Comment:
instead of this I suggest we just do:
`-i ${macbakup} ${TMPPDIR}/cert.cfg`

${macbakup} rigorously w/o quotes and defined only in a previous if that checks if this is a mac machine. The net effect is having the required extension with space on BSD sed, and not having it at all on GNU sed.

I do not even care adding a rm -f at the end, which would have to be conditional to ${macbakup} being defined in order to compose the right file name, although a blanket rm -f *.bak (or whatever the extension) may be acceptable I guess.